app/src/main/java/com/github/wnder/GuessLocationActivity.java

```diff
@@ -299,9 +357,25 @@ private void switchMode() {
         compassMode = !compassMode;
         updateCompassMode();
         if(compassMode) {
+            List<Sensor> list = sensorManager.getSensorList(Sensor.TYPE_ROTATION_VECTOR);
+            if(list.isEmpty()){
+                //We can't use the sensor
+            }
```

Comment:
I'm not sure about what we are doing when the list is empty, maybe just turning back to non-compass mode and/or printing a message would be nice (so that we don't run updateGuessPositionFromHPS uselessely for ex)